### ISSUE TITLE
R Markdown document with shiny runtime doesn't mask functions

### ISSUE THREAD

**Issue by bdforbes** [rstudio/shiny#1876]
I have created a new R Markdown Shiny Document:

````Rmd
---
title: "Untitled"
author: "Ben Forbes"
date: "25 October 2017"
output: html_document
runtime: shiny
---

```{r}
knitr::opts_chunk$set(echo = TRUE)

library(lubridate)
date
```
````

When I run the document, I see the following in the HTML output:

```r
library(lubridate)
## 
## Attaching package: 'lubridate'
## The following object is masked from 'package:base':
## 
##     date
date
## function () 
## .Internal(date())
## <bytecode: 0x0000000017702a68>
## <environment: namespace:base>
```

Clearly, `date` is not being masked in this context. This is presenting issues for me with using the `date` function later in the document. Everything works as expected if I run the code chunk in the local R session or if I remove the `runtime: shiny` from the YAML frontmatter. The problem also occurs with any other masking I have looked at, e.g. from `dplyr`.

Is this behaviour normal? 

Versions:
* R: 3.4.2
* RStudio: 1.1.383
* shiny: 1.0.5
* lubridate: 1.6.0
* knitr: 1.17
* rmarkdown: 1.6

**Comment by bdforbes** [comment]
Interestingly, it works a bit better when the chunk is specifically named `global`, but in a subsequent chunk, the masking still doesn't work:

````Rmd
---
title: "Untitled"
author: "Ben Forbes"
date: "25 October 2017"
output: html_document
runtime: shiny
---

```{r global}
library(lubridate)
date
```

Text

```{r}
ymd_hms
date
```
````

The output is now:

```
library(lubridate)
## 
## Attaching package: 'lubridate'
## The following object is masked from 'package:base':
## 
##     date
date
## function (x) 
## UseMethod("date")
## <environment: namespace:lubridate>
Text

ymd_hms
## function (..., quiet = FALSE, tz = "UTC", locale = Sys.getlocale("LC_TIME"), 
##     truncated = 0) 
## {
##     .parse_xxx_hms(..., orders = c("ymdTz", "ymdT"), quiet = quiet, 
##         tz = tz, locale = locale, truncated = truncated)
## }
## <environment: namespace:lubridate>
date
## function () 
## .Internal(date())
## <bytecode: 0x00000000175d5d30>
## <environment: namespace:base>
```

**Comment by wch** [comment]
I believe this has been fixed by rstudio/rmarkdown#1165. Can you install the latest rmarkdown, restart R, and try again?

**Comment by bdforbes** [comment]
Thanks @wch, it is indeed fixed in the latest GitHub version. Instructions for anyone interested:

```r
install.packages('devtools')
library(devtools)
install_github("rstudio/rmarkdown")
```

Then restart R (can just restart it from within RStudio).
